hudi-cli/src/main/java/org/apache/hudi/cli/commands/ExportCommand.java

```diff
@@ -0,0 +1,152 @@
+ * limitations under the License.
+ */
+
+package org.apache.hudi.cli.commands;
+
+import org.apache.hudi.avro.HoodieAvroUtils;
+import org.apache.hudi.avro.model.HoodieArchivedMetaEntry;
+import org.apache.hudi.cli.HoodieCLI;
+import org.apache.hudi.common.fs.FSUtils;
+import org.apache.hudi.common.model.HoodieLogFile;
+import org.apache.hudi.common.table.log.HoodieLogFormat;
+import org.apache.hudi.common.table.log.HoodieLogFormat.Reader;
+import org.apache.hudi.common.table.log.block.HoodieAvroDataBlock;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.IndexedRecord;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.springframework.shell.core.CommandMarker;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * CLI command to export various information from a HUDI dataset.
+ */
+@Component
+public class ExportCommand implements CommandMarker {
+
+  @CliCommand(value = "export instants", help = "Export Instants and their metadata from the Timeline")
+  public String showArchivedCommits(
+      @CliOption(key = {"limit"}, help = "Limit Instants", unspecifiedDefaultValue = "-1") final Integer limit,
+      @CliOption(key = {"actions"}, help = "Comma seperated list of Instant actions to export",
+        unspecifiedDefaultValue = "clean,commit,deltacommit,rollback,savepoint,restore") final String filter,
+      @CliOption(key = {"desc"}, help = "Ordering", unspecifiedDefaultValue = "false") final boolean descending,
+      @CliOption(key = {"localFolder"}, help = "Local Folder to export to", mandatory = true) String localFolder)
+      throws IOException {
+
+    final String basePath = HoodieCLI.getTableMetaClient().getBasePath();
+    final Path archivePath = new Path(basePath + "/.hoodie/.commits_.archive*");
+    final Path metaPath = new Path(basePath + "/.hoodie/");
+    final Set<String> actionSet = new HashSet<String>(Arrays.asList(filter.split(",")));
+    int numExports = limit == -1 ? Integer.MAX_VALUE : limit;
+    int numCopied = 0;
+
+    if (! new File(localFolder).isDirectory()) {
+      throw new RuntimeException(localFolder + " is not a valid local directory");
+    }
+
+    // The non archived instants are of the format <instantTime>.<action>.<requested/inflight>. We only
+    // want the completed ones which do not have the requested/inflight suffix.
+    FileStatus[] statuses = FSUtils.getFs(basePath, HoodieCLI.conf).listStatus(metaPath);
+    List<FileStatus> nonArchivedStatuses = Arrays.stream(statuses).filter(f -> {
```

Comment:
Can you use the HoodieActiveTimeline here instead to filter out all non requested/inflight. once you do that, the base path to the .hoodie is known to you and the name of the commit file can be gotten from HoodieInstant - then you can do a getFileStatus() if needed - it's cleaner this way rather than you doing indexOf and substring etc - these can break

Reply:
Good idea. It even makes the filtering easier.